src/router/index.js

```diff
@@ -30,21 +27,23 @@ export function syncLocationToStore(store, mountPoints) {
   store.syncedLocationToStore = false;
 }
 
+export const stringifyStoreDataHelper = (data, query = "") => {
+  const cleanQuery = decodeURIComponent(query)
+    .replace(/(^\?)?/, "")
+    .replace(/storeData={.*}&?/, "");
+  const storeDataParam = `storeData=${JSON.stringify(data)}`;
```

Comment:
You should `encodeURIComponent(JSON.stringify(data))` here, just to be sure. That will make it safe for concatenation if it contains an unescaped `&` symbol.